fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkexport/patient/ChunkReader.java

```diff
@@ -0,0 +1,309 @@
+import com.ibm.fhir.model.resource.Resource;
+import com.ibm.fhir.model.util.ModelSupport;
+import com.ibm.fhir.persistence.FHIRPersistence;
+import com.ibm.fhir.persistence.context.FHIRPersistenceContext;
+import com.ibm.fhir.persistence.context.FHIRPersistenceContextFactory;
+import com.ibm.fhir.persistence.helper.FHIRPersistenceHelper;
+import com.ibm.fhir.persistence.helper.FHIRTransactionHelper;
+import com.ibm.fhir.search.compartment.CompartmentUtil;
+import com.ibm.fhir.search.context.FHIRSearchContext;
+import com.ibm.fhir.search.util.SearchUtil;
+
+/**
+ * Bulk patient export Chunk implementation - the Reader.
+ *
+ */
+public class ChunkReader extends AbstractItemReader {
+    private final static Logger logger = Logger.getLogger(ChunkReader.class.getName());
+    int pageNum = 1;
+    int indexOfCurrentResourceType = 0;
+    // Control the number of records to read in each "item".
+    int pageSize = Constants.DEFAULT_SEARCH_PAGE_SIZE;
+
+    private FHIRPersistence fhirPersistence;
+    private List<String> resourceTypes;
+
+    /**
+     * Fhir tenant id.
+     */
+    @Inject
+    @BatchProperty(name = "fhir.tenant")
+    String fhirTenant;
+
+    /**
+     * Fhir data store id.
+     */
+    @Inject
+    @BatchProperty(name = "fhir.datastoreid")
+    String fhirDatastoreId;
+
+    /**
+     * Fhir ResourceType.
+     */
+    @Inject
+    @BatchProperty(name = "fhir.resourcetype")
+    String fhirResourceType;
+
+    /**
+     * Fhir Search from date.
+     */
+    @Inject
+    @BatchProperty(name = "fhir.search.fromdate")
+    String fhirSearchFromDate;
+
+    /**
+     * Fhir search to date.
+     */
+    @Inject
+    @BatchProperty(name = "fhir.search.todate")
+    String fhirSearchToDate;
+
+    /**
+     * Fhir search page size.
+     */
+    @Inject
+    @BatchProperty(name = "fhir.search.pagesize")
+    String fhirSearchPageSize;
+
+    @Inject
+    JobContext jobContext;
+
+    /**
+     * @see AbstractItemReader#AbstractItemReader()
+     */
+    public ChunkReader() {
+        super();
+    }
+
+    private void fillChunkDataBuffer(List<Resource> resources) throws Exception {
+        TransientUserData chunkData = (TransientUserData) jobContext.getTransientUserData();
+        int compartmentPageNum = 1;
+        int resSubTotal = 0;
+        FHIRSearchContext searchContext;
+        Class<? extends Resource> resourceType = ModelSupport
+                .getResourceType(resourceTypes.get(indexOfCurrentResourceType));
+        if (chunkData != null) {
+            for (Resource res : resources) {
+                if (res == null) {
+                    continue;
+                }
+
+                Patient patient = (Patient) res;
+                Map<String, List<String>> queryParameters = new HashMap<>();
+
+                List<String> searchCreterial = new ArrayList<String>();
+
+                if (fhirSearchFromDate != null) {
+                    searchCreterial.add("ge" + fhirSearchFromDate);
```

Comment:
you can link the modifier to fhir-search https://www.hl7.org/fhir/r4/search.html#prefix
highly unlikely to change, but using the String shows  where it's used

Reply:
added the link as comments